src/distdl/nn/mixins/halo_mixin_new.py

```diff
@@ -0,0 +1,100 @@
+import numpy as np
+
+from distdl.utilities.slicing import compute_start_index
+from distdl.utilities.slicing import compute_subshape
+
+
+class HaloMixin:
+
+    def _compute_exchange_info(self,
+                               halo_shape,
+                               neighbor_halo_shapes):
+
+        recv_buffer_shape = halo_shape.copy()
+        send_buffer_shape = np.zeros_like(halo_shape, dtype=int)
+
+        dims = len(halo_shape)
+        for dim in range(dims):
+            left_halo_shape, right_halo_shape = neighbor_halo_shapes[dim]
+
+            if left_halo_shape is not None:
+                send_buffer_shape[dim, 0] = left_halo_shape[dim, 1]
+
+            if right_halo_shape is not None:
+                send_buffer_shape[dim, 1] = right_halo_shape[dim, 0]
+
+        return recv_buffer_shape, send_buffer_shape
+
+    def _left_pad_to_length(self, array, length, value):
+        lpad = length - len(array)
+        return np.pad(array,
+                      pad_width=(lpad, 0),
+                      mode='constant',
```

Comment:
Why `'constant'`?

Reply:
To expand on this, you may want to use a dict similar to the one found in ```distdl.utilities.dtype``` to convert between torch padding modes and numpy padding modes, then pass that value to this function

Reply:
This function just serves to convert the given kernel size, padding, stride and dilations to numpy arrays of proper size. However, since now I see that we only need to do so for padding, so I'll get rid of this function.